File: archived/tests/transpiler/x/pl/map_assign.pl
:- style_check(-singleton).
:- initialization(main).

main :-
    Scores = {alice: 1},
    nth0("bob", Scores, _, T1),
    nth0("bob", Scores1, 2, T1),
    nth0("bob", Scores1, R2), writeln(R2).

File: archived/tests/transpiler/x/java/map_assign.java
public class Main {
    static Data1 scores = new Data1(1);
    static class Data1 {
        int alice;
        Data1(int alice) {
            this.alice = alice;
        }
        boolean containsKey(String k) {
            if (k.equals("alice")) return true;
            return false;
        }
    }


    public static void main(String[] args) {
scores.put("bob", 2);
        System.out.println(scores.get("bob"));
    }
}
